orchestrator/api/heartbeat.py

```
@@ -0,0 +1,216 @@
+        result = await service.run_agent_heartbeat(agent_id)
+        return result
+    except Exception as e:
+        logger.error("Failed to run agent heartbeat for %s: %s", agent_id, e)
+        raise HTTPException(500, f"Heartbeat failed: {str(e)}")
+
+
+@router.get("/agents/{agent_id}/history")
+async def get_agent_heartbeat_history(
+    agent_id: int,
+    limit: int = Query(20, ge=1, le=100),
+    ctx: RequestContext = Depends(get_request_context_hybrid),
```

Ruff (0.15.0)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)